### PR TITLE
Fixes null bugs on WebClient canceled request

### PR DESCRIPTION
**Pull request by adriancole**
There are several places where we didn't check null correctly. This
addresses a couple of them.

### CODE REVIEW

**Comment by violetagg** [comment]
> @violetagg do you know how to trigger `doOnRequestError`? 

Check these tests

https://github.com/reactor/reactor-netty/blob/master/src/test/java/reactor/netty/http/client/HttpClientTest.java#L1110

https://github.com/reactor/reactor-netty/blob/master/src/test/java/reactor/netty/http/client/HttpClientTest.java#L2012

**Comment by adriancole** [comment]
@violetagg any ideas on last commit?

**Comment by adriancole** [comment]
oh I see httpClient and webClient don't share this handler.. only httpClient does

**Comment by adriancole** [comment]
I think I can proceed from here.. thanks for the hint @violetagg!

**Comment by adriancole** [comment]
I was confused as multiple different instrumentation are being tested in a class named "WebClientTests" I'll separate these into separate files in the next commit, so that the next person as easily confused as me doesn't get confused.

**Comment by adriancole** [comment]
moved the `doOnRequestError` failure to https://github.com/spring-cloud/spring-cloud-sleuth/pull/1552

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientBeanPostProcessor.java [hunk 1]
```diff
@@ -374,32 +383,31 @@ void terminateSpanOnCancel() {
...
 								+ this.span + "]");
 					}
-					handleReceive(this.span, clientResponse, throwable);
+					handleReceive(this.span, null, error);
 					return;
 				}
 				int statusCode = clientResponse.rawStatusCode();
-				boolean error = statusCode >= 400;
-				if (error) {
+				boolean isHttpError = statusCode >= 400;
```
Comment (adriancole) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientBeanPostProcessor.java]: ps this isn't correct either, but I'm not changing this now. later we should redo most of this instrumentation

**Pull request merged**